File parsing

j-asefa opened the pull request:
Working ephemeris file parser. There is some error checking still to do and a few test cases to add, but you should be able to work off of this branch and just pass a properly formatted ephemeris file to the `parse_ephemeris_file()` function.

Also, I was having some trouble squashing the commits for this PR, but will try to keep things clean in the future.

ZeyadTamimi commented on kaos/parser.py:
```diff
@@ -0,0 +1,130 @@
+OrbitPoint = namedtuple('OrbitPoint', 'time, pos, vel')
+
+def add_segment_to_db(orbit_data, satellite_id):
+    """Add the given segment to the database.
```
Eh are you aligning to 100 chars, apply comment to entire file :P

ZeyadTamimi commented on kaos/parser.py:
```diff
@@ -0,0 +1,130 @@
+    DB.session.commit()
+
+def parse_ephemeris_file(file_handle):
+    """Parse the given ephemeris file and store
```
Are you aligning to 100 chars?

ZeyadTamimi commented on kaos/parser.py:
```diff
@@ -0,0 +1,130 @@
+
+    DB.session.commit()
+
+def parse_ephemeris_file(file_handle):
```
Not really a file handle, its literally a path

ZeyadTamimi commented on kaos/parser.py:
```diff
@@ -0,0 +1,130 @@
+    DB.session.commit()
+
+    with open(file_handle, "rU") as f:
+        """A list containing each of the 15 or so
```
Comments as to what the variable is should not be there. The name should make it obvious ( which it does )

ZeyadTamimi commented on kaos/parser.py:
```diff
@@ -0,0 +1,130 @@
+            if read_orbital_data:
+                line = line.strip()
+                if line:
+                    """each row is a 7-tuple formatted as
```
Should probably be int he doc of the function since its basically  precondition on the file. Unsure

ZeyadTamimi commented on kaos/parser.py:
```diff
@@ -0,0 +1,130 @@
+                    ended with. So we don't want to commit the first entry in a
+                    *new* segment. The last_seen_segment_boundary != orbit_tuple.time checks
+                    this and makes sure not to commit to the db in this case."""
+                    if orbit_tuple.time in segment_boundaries:
```
And these two conditions

ZeyadTamimi commented on kaos/parser.py:
```diff
@@ -0,0 +1,130 @@
+                        if last_seen_segment_boundary != orbit_tuple.time:
+                            last_seen_segment_boundary = orbit_tuple.time
+                            add_segment_to_db(segment_tuples, sat.platform_id)
+                            del segment_tuples[:]
```
Not sure about using del, just set to empty segment_tuples = []

ZeyadTamimi commented on test/test_parser.py:
```diff
@@ -0,0 +1,26 @@
+    """Ensures that the response history table behaves as expected."""
+
+    def test_simple_parser_J2000(self):
+        parse_ephemeris_file("ephemeris/Radarsat2_J2000.e")
```
Test number of segments, and the min max time!

ZeyadTamimi commented on test/test_parser.py:
```diff
@@ -0,0 +1,26 @@
+        self.assertTrue(len(orbit.query.all()) == 17307) #taken from ephem file
+
+    def test_simple_parser_Fixed(self):
+        parse_ephemeris_file("ephemeris/Radarsat2_Fixed.e")
```
Ditto

j-asefa commented:
@ZeyadTamimi I think I addressed all of your concerns. There is one test case that is broken. It's under `test_ephemeris_parser_single_file` in `test_parser.py`. 

This test fails at `self.assertTrue(q.start_time == seg_start)`, and I'm not sure why. I've printed both values extensively and they seem to be the exact same. I'm at a loss as to why it's failing.

j-asefa commented on test/test_parser.py:
```diff
@@ -1,26 +1,147 @@
+	query = orbit_segment.query.all()
+
+	for q, seg_start in zip(query, seg_times[:-1]):
+	    self.assertTrue(q.start_time == seg_start)
```
This check is failing and I'm not sure why.

ZeyadTamimi replied:
I have an idea, those two values are floats right? You need a fuzzy compare. Ill try to look at it today.

rouyeen replied:
maybe assertAlmostEqual() ?

ZeyadTamimi commented:
test_db_add_correct_num_rows is still flaky

Also dont import import *

ZeyadTamimi commented:
In the interest of getting people actual test data, I am merging this change :+1:
